ui/src/app/components/sidenav-rule/sidenav-rule.component.ts

```diff
@@ -52,7 +52,7 @@ export class SidenavRuleComponent implements OnInit {
     this.ruleForm.disable()
     this.ruleForm.controls['ruleName'].setValue(data?.Name)
     this.ruleForm.controls['ruleType'].setValue(
-      this.viewRuleData?.Type === 'add_index' ? 'addIndex' : 'globalDataType'
+      this.viewRuleData?.Type === 'add_index' ? 'addIndex' : (this.viewRuleData?.Type === 'global_datatype_change' ? 'globalDataType' : 'changeMaxLength')
```

Comment:
can there only be 2 rules here ? If not, a switch case might be more appropriate here.